src/useHookFormMask.ts

```diff
@@ -6,13 +6,11 @@ import {
 import { flow, getMaskOptions } from './utils';
 import { Mask, Options } from './types';
 
-export function useHookFormMask<
-  T extends FieldValues, D extends RegisterOptions,
->(registerFn: UseFormRegister<T>) {
-  return (fieldName: Path<T>, mask: Mask, options?: Options & D) => {
+export function useHookFormMask<T extends FieldValues>(registerFn: UseFormRegister<T>) {
+  return (fieldName: Path<T>, mask: Mask, options?: Options, registerOptions?: RegisterOptions) => {
     if (!registerFn) throw new Error('registerFn is required');
 
-    const { ref, ...restRegister } = registerFn(fieldName);
+    const { ref, ...restRegister } = registerFn(fieldName, registerOptions);
```

Comment:
I really dont like to use 3 params funcion. Not provide an good development experience. The secound funcion are an merge from Inputmask options and hook-form options. We dont need that.

Reply:
I see but unless I'm missing something, I think you're merging only the types, and the params belonging to `react-hook-form` register are never really passed to it?

What I mean is, for example, it will auto complete options like `setValueAs`, `valueAsNumber`, `valueAsDate`, which belong to register, but if if pass them, it will have no effect.

with the original code (after we fix the types import), TS won't complain about this

```ts
{...registerWithMask(
  'price',
  'brl-currency',
  { rightAlign: false, autoUnmask: true, setValueAs: (val) => (val.replace(',', '.')) }
)}
```

but the `setValueAs` will have no effect

with the modified code, we could have something like

```ts
{...registerWithMask(
  'price',
  'brl-currency',
  { rightAlign: false, autoUnmask: true },
  { setValueAs: (val) => (val.replace(',', '.')) }
)}
```

It was the simplest way I could think without messing a lot with the original code.

Now let me explain how I use this specificaly. When I use a schema like `Joi.number()` for my `price` field, it validates numbers and strings that can be converted to numbers directly. So doing this is the simplest way to keep my schemas happy when I use commas for decimals, etc. I could also write a custom Joi validation, but passing `setValueAs` to register is simpler.

I guess what I don't understand is why join the options if you can't do anything with the joined options?

Reply:
What if we keep the merged types and pass only `setValueAs` to `registerFn` for now?

```ts
const { setValueAs } = options || {};

const { ref, ...restRegister } = registerFn(fieldName, { setValueAs });
```

Most of the other `RegisterOptions` we can pass direct to the component anyway.

I also tried passing `valueAsNumber` and `valueAsDate` and TS started yelling at me, but I'm ok converting to Number inside onSubmit, etc.

Reply:
Uh, check my new change request.